lmdeploy/serve/turbomind/triton_models/postprocessing/1/model.py

```diff
@@ -2,88 +2,11 @@
-
-    Args:
-        model_file (str): the path of the tokenizer model
-    """
-
-    def __init__(self, model_file: str):
-        model_folder = osp.split(model_file)[0]
-        tokenizer_config_file = osp.join(model_folder, 'tokenizer_config.json')
-        use_hf_model = osp.exists(tokenizer_config_file)
-        self.use_hf_model = use_hf_model
-        if not self.use_hf_model:
-            from sentencepiece import SentencePieceProcessor
-            self.model = SentencePieceProcessor(model_file=model_file)
-            self.vocab_size = self.model.vocab_size()
-            self.start_id = self.model.bos_id()
-            self.end_id = self.model.eos_id()
-        else:
-            from transformers import AutoTokenizer
-            backend_tokenizer_file = osp.join(model_folder, 'tokenizer.json')
-            if not osp.exists(backend_tokenizer_file):
-                print('WARNING: Can not find tokenizer.json. '
-                      'It may take long time to initialize the tokenizer.')
-            self.model = AutoTokenizer.from_pretrained(model_folder,
-                                                       trust_remote_code=True)
-            self.vocab_size = self.model.vocab_size
-            self.start_id = self.model.bos_token_id
-            self.end_id = self.model.eos_token_id
-            # save tokenizer.json to reuse
-            if not osp.exists(backend_tokenizer_file) and \
-                    hasattr(self.model, 'backend_tokenizer'):
-                self.model.backend_tokenizer.save(backend_tokenizer_file)
-
-    def encode(self, s: str):
-        """Tokenize a prompt.
-
-        Args:
-            s (str): a prompt
-        Returns:
-            list[int]: token ids
-        """
-        if not self.use_hf_model:
-            add_bos = False
-            add_eos = False
-            if s.find('<BOS>') != -1:
-                s = s.replace('<BOS>', '')
-                add_bos = True
-            if s == '<EOS>':
-                s = ''
-                add_eos = True
-            return self.model.Encode(s, add_bos=add_bos, add_eos=add_eos)
-        else:
-            add_special_tokens = False
-            if s.find('<BOS>') != -1:
-                s = s.replace('<BOS>', '<s>')
-            if s == '<EOS>':
-                s = '</s>'
-            if len(s) == 0:
-                add_special_tokens = True
-            return self.model.encode(s, add_special_tokens=add_special_tokens)
-
-    def decode(self, t: List[int]):
-        """De-tokenize.
-
-        Args:
-            t (List[int]): a list of token ids
-        Returns:
-            str: text of decoding tokens
-        """
-        if not self.use_hf_model:
-            return self.model.Decode(t)
-        else:
-            skip_special_tokens = False
-            return self.model.decode(t,
-                                     skip_special_tokens=skip_special_tokens)
+from .tokenizer.tokenizer import Tokenizer
```

Comment:
why not import from `lmdeploy.tubomind.tokenizer`? User can not find such a Tokenizer before we copy file.

Reply:
FIrst of all, we have too many tokenizers now, preprocessing/1/model.py, postprocessing/1/model.py and turbomind/tokenizer.py. It is not conducive to maintaining the code.

Secondly, preprocessing and postprocessing are used internally by triton inference server, not API. If user want to serve a model based on triton inference server, they have to convert the model first by running `python deploy.py`. Then turbomind's tokenizer.py will be copied to both preprocessor and postprocessor

I will update comments to clear it up